include/deal.II/lac/trilinos_tpetra_sparse_matrix.templates.h

```diff
@@ -1005,6 +1005,68 @@ namespace LinearAlgebra
 
 
 
+    template <typename Number, typename MemorySpace>
+    void
+    SparseMatrix<Number, MemorySpace>::clear_row(const size_type row,
+                                                 const Number    new_diag_value)
+    {
+      clear_rows(std::vector<size_type>(1, row), new_diag_value);
+    }
+
+
+
+    template <typename Number, typename MemorySpace>
+    void
+    SparseMatrix<Number, MemorySpace>::clear_rows(
+      const std::vector<size_type> &rows,
```

Comment:
Perhaps make this second argument an `ArrayView`? Because `std::vector` can implicitly be converted to `ArrayView`, this is only a minor incompatibility to the Epetra wrappers, but it allows you to not have to allocate memory in line 1013 above where you just forward to the other function.

Reply:
Sure, let me do that consistently through the whole code base so we have an uniform interface.

Reply:
That's ok. I'd also be ok if you wanted to leave that to a follow-up.

Reply:
I see you're already way ahead of me :-)